academic_observatory_workflows/workflows/unpaywall_data_feed_telescope.py

```diff
@@ -0,0 +1,326 @@
+        :param start_date: the start_date of the release.
+        :param end_date: the end_date of the release.
+        :param first_release: whether this is the first release that is processed for this DAG
+        """
+
+        super().__init__(
+            dag_id,
+            start_date,
+            end_date,
+            first_release,
+        )
+
+        self.http_header = get_observatory_http_header(package_name="academic_observatory_workflows")
+
+    @property
+    def api_key(self) -> str:
+        """The API key for accessing Unpaywall."""
+
+        return get_airflow_connection_password(UnpaywallDataFeedRelease.AIRFLOW_CONNECTION)
+
+    @property
+    def snapshot_url(self) -> str:
+        """Snapshot URL"""
+
+        return f"{UnpaywallDataFeedRelease.SNAPSHOT_URL}?api_key={self.api_key}"
+
+    @property
+    def data_feed_url(self) -> str:
+        """Data Feed URL"""
+
+        return f"{UnpaywallDataFeedRelease.CHANGEFILES_URL}?interval=day&api_key={self.api_key}"
+
+    def download(self):
+        """Download the release."""
+        # May need to refactor depending on how snapshot looks.
+        if self.first_release:
+            self.download_snapshot_()
+        else:
+            self.download_data_feed_()
+
+    def download_snapshot_(self):
+        """Download the most recent Unpaywall snapshot on or before the start date."""
+
+        download_file(url=self.snapshot_url, headers=self.http_header, prefix_dir=self.download_folder)
+        download_date = UnpaywallRelease.parse_release_date(self.download_files[0]).date()
+        start_date = self.start_date.date()
+
+        if start_date != download_date:
+            raise AirflowException(
+                f"The telescope start date {start_date} and the downloaded snapshot date {download_date} do not match."
+            )
+
+    @staticmethod
+    def get_diff_release(*, feed_url: str, start_date: pendulum.DateTime) -> Tuple[Union[None, str]]:
+        """Get the differential release url and filename.
+
+        :param feed_url: The URL to query for releases.
+        :param start_date: Earliest date to consider.
+        :return: (None,None) if nothing found, otherwise (url, filename).
+        """
+
+        release_info = get_http_response_json(feed_url)
+        for release in release_info["list"]:
+            # Have been advised by Unpaywall to parse timestamp from filename instead of relying on the json fields.
+            release_date = UnpaywallRelease.parse_release_date(release["filename"]).date()
+
+            # Apply diffs from 2 days ago.  This is so we start applying diffs 1 day before the snapshot date to
+            # guarantee no gaps with the snapshot.
+            target_date = (start_date - pendulum.Duration(days=2)).date()
+
+            if release_date == target_date:
+                return release["url"], release["filename"]
+
+        return (None, None)
+
+    def download_data_feed_(self):
```

Comment:
Private methods in Python should begin with an underscore, not end with an underscore, e.g. _download_snapshot.